In-house removal of PyECLib in Swift since we do not support the
Erasure Code storage policy.

This patch is Solaris-specific and not suitable for upstream.

--- swift-2.7.0/swift/common/storage_policy.py.~1~	2016-03-25 02:42:55.000000000 -0700
+++ swift-2.7.0/swift/common/storage_policy.py	2016-08-08 19:02:28.033539550 -0700
@@ -22,7 +22,12 @@ from swift.common.utils import (
 from swift.common.ring import Ring, RingData
 from swift.common.utils import quorum_size
 from swift.common.exceptions import RingValidationError
-from pyeclib.ec_iface import ECDriver, ECDriverError, VALID_EC_TYPES
+
+# Attempt to load PyEClib if installed
+try:
+    from pyeclib.ec_iface import ECDriver, ECDriverError, VALID_EC_TYPES
+except ImportError:
+    pass
 
 LEGACY_POLICY_NAME = 'Policy-0'
 VALID_CHARS = '-' + string.ascii_letters + string.digits
--- swift-2.7.0/swift/proxy/controllers/obj.py.~1~	2016-03-25 02:42:55.000000000 -0700
+++ swift-2.7.0/swift/proxy/controllers/obj.py	2016-08-08 19:02:28.035065785 -0700
@@ -65,7 +65,7 @@ from swift.common.http import (
     HTTP_INSUFFICIENT_STORAGE, HTTP_PRECONDITION_FAILED, HTTP_CONFLICT,
     HTTP_UNPROCESSABLE_ENTITY, HTTP_REQUESTED_RANGE_NOT_SATISFIABLE)
 from swift.common.storage_policy import (POLICIES, REPL_POLICY, EC_POLICY,
-                                         ECDriverError, PolicyError)
+                                         PolicyError)
 from swift.proxy.controllers.base import Controller, delay_denial, \
     cors_validation, ResumingGetter
 from swift.common.swob import HTTPAccepted, HTTPBadRequest, HTTPNotFound, \
@@ -76,6 +76,12 @@ from swift.common.swob import HTTPAccept
 from swift.common.request_helpers import is_sys_or_user_meta, is_sys_meta, \
     remove_items, copy_header_subset
 
+# Load PyEClib if installed
+try:
+    from swift.common.storage_policy import ECDriverError
+except ImportError:
+    pass
+
 
 def copy_headers_into(from_r, to_r):
     """
